pydantic_ai_slim/pydantic_ai/messages.py

```diff
@@ -445,3 +446,30 @@ class PartDeltaEvent:
 
 ModelResponseStreamEvent = Annotated[Union[PartStartEvent, PartDeltaEvent], pydantic.Discriminator('event_kind')]
 """An event in the model response stream, either starting a new part or applying a delta to an existing one."""
+
+
+@dataclass
+class FunctionToolCallEvent:
+    """An event indicating the start to a call to a function tool."""
+
+    part: ToolCallPart
+    """The (function) tool call to make."""
+    call_id: uuid.UUID = field(default_factory=uuid.uuid4, repr=False)
```

Comment:
what if the model provider returns a call id, should we use `str` here to be more lenient?

Reply:
We could; I was a bit nervous to use the vendor-provided call_id even if it was present because, for all I know, some vendors might use a value that isn't unique. The point here was purely to be able to have a consistent ID to be able to match up the request to the tool call and the result from it; I can see why it might be nice to use the same value provided by the vendor if one is provided by the vendor, but I just don't know how much I can assume about that in general.

Reply:
I have modified this to be of type str, and use the part's tool_call_id if it is present (otherwise it uses `str(uuid.uuid4())`.